### PR TITLE
Refactoring account/domain validation webhook logic

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
As I understand, the criteria should be that if account is provided, then domain must be as well. Otherwise, domain is optional regardless of account. This PR attempts to modify the current logic to align with that. 

I encountered this issue after adding the "account" field and trying to spin up a cluster, only specifying domain (as I had been doing prior to this change)

*Testing performed:*
make build && make test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rejoshed — APPROVED** [review]

File: api/v1alpha3/cloudstackcluster_webhook.go [hunk 1]
```diff
@@ -62,8 +62,8 @@ func (r *CloudStackCluster) ValidateCreate() error {
 		errorList = append(errorList, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
-	if (r.Spec.Account == "") != (r.Spec.Domain == "") {
-		errorList = append(errorList, field.Required(field.NewPath("spec", "account"), "account and domain must be specified together"))
+	if (r.Spec.Account != "") && (r.Spec.Domain == "") {
```
Comment (jweite-amazon) [inline api/v1alpha3/cloudstackcluster_webhook.go:65]: Why the change?  The original construct is surprising, but effectively saying that if account == "" is true then domain must == "" must also be true, and if account == "" is false then domain must == "" must also be false.

The changed logic doesn't detect domain provided but account blank.

Comment (jweite-amazon) [reply api/v1alpha3/cloudstackcluster_webhook.go:65]: I just read the PR description.  I hadn't understood it quite they way you described it.  Let me think on this a minute...

Comment (jweite-amazon) [reply api/v1alpha3/cloudstackcluster_webhook.go:65]: I stand corrected.  The API Ref states that domain must be provided if account is provided, but nothing about the other way around.
